### DISCUSSION TITLE
Which feature do you want in the new Version 3 (future3)?

### DISCUSSION THREAD

**Discussion by s-martin** [MiczFlor/RPi-Jukebox-RFID#2172] [Polls]
The new [version 3 (future3)](https://github.com/MiczFlor/RPi-Jukebox-RFID/tree/future3/main/documentation) of Phoniebox is becoming mature and will soon be the new default of Phoniebox.

Version 3 still misses a few features (see [feature status](https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/future3/main/documentation/developers/status.md)).

To get a feeling, which missing features may be the most wanted, please vote in this poll.

If there’s a feature not listed, please add a comment.

Unfortunately multiple votes are not possible, so please add your other votes in comments (or upvote existing comments)!

Thanks for your help.

**Comment by s-raeuchle** [comment]
- Tidal Support :)

**Comment by sosi-deadeye** [comment]
I'm not a user of this project, but I think the addition of MQTT could make it easier for customization.

**Comment by TheoRK667** [comment]
Jellyfin support for my collection of music on my NAS

**Comment by Alameleon** [comment]
I wish a Deezer support (it`s similar like spotify).

Ich wünsche mir die Unterstützung für Deezer. Ist wie Spotify. Aber ich habe einen Deezer Acc durch Vodafone kostenlos bekommen damals und daher bin ich da trotz Daten-Skandal geblieben. Würde bestimmt zu Spotify wechseln, aber ALLES jetzt umändern bezüglich Lieblingsmusik, Interpreten, Playlists, etc. ist mir zu astrengend.

**Comment by Mannshoch** [comment]
I wish an ultra fast boot of the device. It shouldbe possible that powerless is standard.

**Reply by Mannshoch** [reply]
If it's possible an energy saving mode like on desktop where a tip onto the keyboard boot the pc may be nice. The moment I place a card onto the box, it wakes.

The base of phoniebox is a raspbian. May an ultra slim mode could be helping, where every service is stopped that is not needed for the configured phoniebox and they start only if a certain card is swiped. 

May Phoniebox could create it's own systemd target 'phoniebox.target' where only phonybox related services are running. 
On the phoniebox website you could setup a internet ON/OFF swipe card

**Reply by Mannshoch** [reply]
comment updated.

**Comment by TomK32** [comment]
Here's two things I'd like to add myself, hints where to start would be nice:

- Allow cards to continue playing where a track was stopped last time. This would be fantastic for audiobooks
- Play xml/m3u feeds, the use case here is the BR Betthupferl that my daughter loves. Playing the N most recent entries would be ideal. No need to store the files, though that might be a nice extra option

**Reply by AlvinSchiller** [reply]
Hi, 
i think both cases are already tracked. Feel free to join the discussions
* resume option https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/1946
* playlists https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2200

**Comment by s-martin** [comment]
For anyone interested in podcasts the feature is already available, but the documentation is not up to date yet.

See here for docs: https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/b7b29a2a524d03af509340a9d6025e4b6b4413ff/documentation/builders/webapp/playlists-livestreams-podcasts.md

Belongs to this PR https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2200

**Edit:**
docs are now up to date: https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/future3/main/documentation/builders/webapp/playlists-livestreams-podcasts.md

**Comment by chiefbrain** [comment]
I like to use the mcp23017 port expander to be able to use more buttons/sensors/leds.
WiringPi has support for this expander whereas gpiozero or pigpio havent.

**Reply by s-martin** [reply]
could be possible 

See my answer in #1234

**Comment by m-a-johnson** [comment]
Would love it if I could play videos. My goal is to have a device that my kids can swipe an RFID card, and have a video come up with a specific loved one reading them a book (i.e. each loved one pre-records a video of them reading a book, and then have it play upon request so they can 'read-along'). Then, when no books are being played, maybe just have a looped video or static photo as a background.

**Comment by mrsgrobi** [comment]
Ich habe zwei Vorschläge und bringe die mal auf deutsch. Ich hoffe, das ist i.O.:

1. Ich wünsche mir, dass die Status-LED schon leuchtet, während der raspi hochfährt und blinkt, sobald die Phoniebox betriebsbereit ist. Ich habe das provisorisch selbst gelöst mit z.B. "gpio=24=op,dh" in der config.txt und dann in der rc.config ein Skript gestartet, das das Blinken übernimmt.
2. Wenn man ein Relaismodul hat, das per GPIO geschaltet werden kann, kann man auch das "Einschaltproblem" ohne weitere Zusätze lösen. Einfach die gesamte Stromversorgung über den Schalter des Relais führen und das per Einschalttaster überbrücken. Dann beginnt der raspi beim Druck auf den Taster zu starten und schaltet (ähnlich wie die LED) einen (anderen) GPIO ein, der dann das Relais anziehen lässt. Auch das habe ich erfolgreich so umgesetzt.

Bei Verständnisproblemen (der Anforderungen und der von mir gewählten Realisierung) unterstütze ich gerne.

**Comment by 3dprintedmac** [comment]
I'd still love to update the version three from version two, but the one feature I'm still waiting for is the ability to control the playlist with individual RFID tags. We didn't put a wired buttons on our player for simplicity, and it's nice for my toddler tohave a card, they scan to move to the next song when they want to.

**Comment by funetn** [comment]
Would like to see Tidal (Connect) integration made more seamless/streamlined.  I can get Tidal-Connect working with Mopid but not this app.  With ppl paying for music subscriptions 1) it's just more convenient to point to online streaming services for songs/albums to play; 2) eliminates the need to rip music thus saving space; 3) unlimited source of music to point to; 4) additional perk is that artists get paid per listen (which they would anyway if listening via the Tidal app).

**Comment by mtill** [comment]
I'd prefer to have a more lightweight setup with minimal boot time that will also work on a raspberry pi zero.
I implemented a minimalist alternative here: https://github.com/mtill/rfid-music-player
... might be useful for discussion / possible later integration ...

**Comment by funetn** [comment]
DAAP-FORKED or Owntone integration to allow for wireless transmitting to Airplay, Chromecast, Roku, etc. devices.  (per their Github - iTunes (DAAP), Roku (RSP), AirPlay devices (multi-room), Chromecast and also supports local playback).  I was able to get Owntone installed on my Phoniebox/MOPIDY/RFID-Music-Player RPi and to stream to an Airplay receiver.  However, I lost the ability to read RFID cards for some reason.  Still trying to figure out why that is (the hw still shows in i2cdetect).

**Comment by EarMaster** [comment]
Uploading files via the web interface would be nice. It is something I always used on v2 and really miss on v3.

**Reply by pabera** [reply]
We are tracking this here: #2138. I am looking into it.

**Comment by bvanhee** [comment]
Two audio sources I'd like to have:

- youtube
- bluetooth: it would be convenient to simply stream from my phone to the phoniebox, especially for "one time listens". [moode](https://github.com/moode-player/moode) does that and it's based on MPD so it should be feasible on v3
